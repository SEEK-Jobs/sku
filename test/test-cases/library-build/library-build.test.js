const path = require('path');
const dirContentsToObject = require('../../utils/dirContentsToObject');
const waitForUrls = require('../../utils/waitForUrls');
const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const { getAppSnapshot } = require('../../utils/appSnapshot');
const appDir = path.resolve(__dirname, 'app');
const distDir = path.resolve(appDir, 'dist');

describe('library-build', () => {
  describe('build', () => {
    beforeAll(async () => {
      await runSkuScriptInDir('build', appDir);
    });

    it('should generate the expected files', async () => {
      const files = await dirContentsToObject(distDir);
      expect(files).toMatchSnapshot();
    });
  });

  describe('start', () => {
    const devServerUrl = `http://localhost:8082`;
    let server;

    beforeAll(async () => {
      server = await runSkuScriptInDir('start', appDir);
      await waitForUrls(devServerUrl);
    });

    afterAll(() => {
      server.kill();
    });

    it('should start a development server', async () => {
      const snapshot = await getAppSnapshot(devServerUrl);
      expect(snapshot).toMatchSnapshot();
    });
  });
});
