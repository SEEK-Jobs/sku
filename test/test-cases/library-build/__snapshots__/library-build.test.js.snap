// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`library-build build should generate the expected files 1`] = `
Object {
  "MyLibrary.css": "._1ys27hL{color:red}
",
  "MyLibrary.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;

exports[`library-build start should start a development server 1`] = `
Object {
  "content": "<!DOCTYPE html><html><head>
    <meta charset=\\"UTF-8\\">
    <title>My Awesome Project</title>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/MyLibrary.css\\">
  </head>
  <body>
    <script type=\\"text/javascript\\" src=\\"/MyLibrary.js\\" defer=\\"\\"></script>
    <script>window.MyLibrary();</script>
  
</body></html>",
  "errors": Array [],
  "sourceHtml": "<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"UTF-8\\">
    <title>My Awesome Project</title>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/MyLibrary.css\\" />
  </head>
  <body>
    <script type=\\"text/javascript\\" src=\\"/MyLibrary.js\\" defer></script>
    <script>window.MyLibrary();</script>
  </body>
</html>",
  "warnings": Array [],
}
`;
